### PR TITLE
Intel 471 feed

### PR DESCRIPTION
**Pull request by JasBeilin**
## Status
- [x] In Progress

## Related Issues
fixes: https://github.com/demisto/etc/issues/25106

## Description
Adding Malware feed and actor feed

## Minimum version of Demisto
- [X] 5.5.0

## Does it break backward compatibility?
   - [X] No

## Must have
- [X] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/FeedIntel471/Integrations/Intel471Actors/Intel471Actors.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
+from typing import Dict, List
+
+import demistomock as demisto
+from CommonServerPython import *
+from JSONFeedApiModule import *  # noqa: E402
+
+# from Packs.ApiModules.Scripts.JSONFeedApiModule.JSONFeedApiModule import *
+
```
Comment (bakatzir) [inline Packs/FeedIntel471/Integrations/Intel471Actors/Intel471Actors.py:8]:
```suggestion
```

File: Packs/FeedIntel471/Integrations/Intel471Actors/Intel471Actors.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    This function gets parameters and adding the relevant to a url string
+    """
+    url_suffix = ""
+    if 'actor' not in kwargs:
+        kwargs['actor'] = '*'
+
+    for param in kwargs:
+        if param in SEARCH_PARAMS:
+            url_suffix += f"&{SEARCH_PARAMS.get(param)}={kwargs.get(param)}"
+    return FEED_URL + url_suffix.strip('&')
```
Comment (bakatzir) [inline Packs/FeedIntel471/Integrations/Intel471Actors/Intel471Actors.py:38]:
```suggestion
    return f'{FEED_URL}{url_suffix.strip('&')}'
```

File: Packs/FeedIntel471/Integrations/Intel471Actors/Intel471Actors.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+
+def custom_build_iterator(client: Client, feed: Dict, limit: int = 0, **kwargs) -> List:
+    """
+    This function replace the build iterator function in JsonFeedApiModule in order to enable paging specific to api.
+    Paginf is done using
```
Comment (bakatzir) [inline Packs/FeedIntel471/Integrations/Intel471Actors/Intel471Actors.py:44]: ?
```suggestion
    Paginf is done using
```

File: Packs/FeedIntel471/Integrations/Intel471Malware/Intel471Malware.py [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+INDICATOR_VALUE_FIELD = {FeedIndicatorType.File: 'indicator_data_file_sha256',
+                         FeedIndicatorType.URL: 'indicator_data_url',
+                         "ipv4": 'indicator_data_address'}
+
+
+def _create_url(**kwargs):
+    url_suffix = ""
+    for param in kwargs:
+        url_suffix += f"&{param}={kwargs.get(param)}"
+    return FEED_URL + url_suffix.strip('&')
```
Comment (bakatzir) [inline Packs/FeedIntel471/Integrations/Intel471Malware/Intel471Malware.py:57]:
```suggestion
    return f'{FEED_URL}{url_suffix.strip('&')}'
```

File: Packs/FeedIntel471/Integrations/Intel471Malware/Intel471Malware.yml [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+  required: false
+  type: 0
+- additionalinfo: How far back in time to go when performing the first fetch
+  defaultvalue: '7 days'
+  display: First fetch timestamp (<number> <time unit>, e.g., 12 hours, 7 days)
+  hidden: false
+  name: fetch_time
+  required: false
+  type: 0
+- additionalinfo: 'Long unix time. Search indicators with activity range overlapping given time (including). Object fields: activity.first and activity.last. Empty indicates unbounded'
```
Comment (bakatzir) [inline Packs/FeedIntel471/Integrations/Intel471Malware/Intel471Malware.yml:111]: add an example.

File: Packs/FeedIntel471/Integrations/Intel471Malware/Intel471Malware.yml [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+  display: Start Time
+  name: from
+  required: false
+  type: 0
+- additionalinfo: 'Long unix time. Search indicators with activity range overlapping given time (including). Object fields: activity.first and activity.last. Empty indicates unbounded'
```
Comment (bakatzir) [inline Packs/FeedIntel471/Integrations/Intel471Malware/Intel471Malware.yml:116]: add an example.

File: Packs/FeedIntel471/Integrations/Intel471Malware/Intel471Malware.yml [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+  required: false
+  type: 0
+- display: Trust any certificate (not secure)
+  name: insecure
+  required: false
+  type: 8
+- display: Use system proxy settings
+  name: proxy
+  required: false
+  type: 8
```
Comment (bakatzir) [inline Packs/FeedIntel471/Integrations/Intel471Malware/Intel471Malware.yml:144]: these 2 should be at the bottom

File: Packs/FeedIntel471/Integrations/Intel471Actors/Intel471Actors.yml [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+  required: false
+  type: 0
+- display: Trust any certificate (not secure)
+  name: insecure
+  required: false
+  type: 8
+- display: Use system proxy settings
+  name: proxy
+  required: false
+  type: 8
```
Comment (bakatzir) [inline Packs/FeedIntel471/Integrations/Intel471Actors/Intel471Actors.yml:115]: these 2 should be at the bottom

File: Packs/FeedIntel471/Integrations/Intel471Actors/Intel471Actors.yml [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+  required: false
+  type: 0
+- additionalinfo: How far back in time to go when performing the first fetch
+  defaultvalue: '7 days'
+  display: First fetch timestamp (<number> <time unit>, e.g., 12 hours, 7 days)
+  hidden: false
+  name: fetch_time
+  required: false
+  type: 0
+- additionalinfo: 'Long unix time. Search indicators with activity range overlapping given time (including). Object fields: activity.first and activity.last. Empty indicates unbounded'
```
Comment (bakatzir) [inline Packs/FeedIntel471/Integrations/Intel471Actors/Intel471Actors.yml:82]: add an example.

File: Packs/FeedIntel471/Integrations/Intel471Actors/Intel471Actors.yml [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+  display: Start Time
+  name: from
+  required: false
+  type: 0
+- additionalinfo: 'Long unix time. Search indicators with activity range overlapping given time (including). Object fields: activity.first and activity.last. Empty indicates unbounded'
```
Comment (bakatzir) [inline Packs/FeedIntel471/Integrations/Intel471Actors/Intel471Actors.yml:87]: add an example.

File: Packs/FeedIntel471/Integrations/Intel471Actors/Intel471Actors.yml [hunk 11]
```diff
@@ -0,0 +1,157 @@
...
+      defaultValue: '50'
+      description: The maximum number of results to return. The default value is 50.
+      isArray: false
+      name: limit
+      required: false
+      secret: false
+    deprecated: false
+    description: Gets the feed indicators.
+    execution: false
+    name: intel471-get-indicators
```
Comment (bakatzir) [inline Packs/FeedIntel471/Integrations/Intel471Actors/Intel471Actors.yml:145]: we do not want the same cmd to be in 2 different integrations. please rename to `intel471(-)actors-get-indicators`

File: Packs/FeedIntel471/Integrations/Intel471Malware/Intel471Malware.yml [hunk 12]
```diff
@@ -0,0 +1,186 @@
...
+      defaultValue: '50'
+      description: The maximum number of results to return. The default value is 50.
+      isArray: false
+      name: limit
+      required: false
+      secret: false
+    deprecated: false
+    description: Gets the feed indicators.
+    execution: false
+    name: intel471-get-indicators
```
Comment (bakatzir) [inline Packs/FeedIntel471/Integrations/Intel471Malware/Intel471Malware.yml:174]: we do not want the same cmd to be in 2 different integrations. please rename to `intel471(-)malware-get-indicators`

File: Packs/ApiModules/Scripts/JSONFeedApiModule/JSONFeedApiModule.py [hunk 13]
```diff
@@ -130,47 +130,78 @@ def fetch_indicators_command(client: Client, indicator_type: str, feedTags: list
...
-                'value': indicator_value,
-                'type': current_indicator_type,
-                'fields': {
-                    'tags': feedTags,
-                }
-            }
 
-            if client.tlp_color:
-                indicator['fields']['trafficlightprotocol'] = client.tlp_color
+            handle_indicator_function(client, item, feed_config, service_name, indicator_type, indicator_field,
```
[the remark this answers is not in the document]
Comment (JasBeilin) [reply Packs/ApiModules/Scripts/JSONFeedApiModule/JSONFeedApiModule.py]: its the name of the function given in the cofiguration. I can make the default one "private"

**Comment by ShirleyDenkberg** [comment]
@bakatzir  Doc review completed.

**Comment by bakatzir** [comment]
Waiting on default reputation and reliability of both feeds prior to the merge.

File: Packs/FeedAWS/ReleaseNotes/1_1_1.md [hunk 14]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### AWS Feed
+Infrastructure improvements.
```
Comment (bakatzir) [inline Packs/FeedAWS/ReleaseNotes/1_1_1.md:4]:
```suggestion
- Internal code improvements.
```

File: Packs/FeedFastly/ReleaseNotes/1_1_1.md [hunk 15]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### Fastly Feed
+Infrastructure improvements.
```
Comment (bakatzir) [inline Packs/FeedFastly/ReleaseNotes/1_1_1.md:4]:
```suggestion
- Internal code improvements.
```

File: Packs/FeedIntel471/Integrations/Intel471Actors/Intel471Actors_test.py [hunk 16]
```diff
@@ -0,0 +1,37 @@
+import pytest
+import Intel471Actors as feed
+
+# import demistomock as demisto
+
```
Comment (bakatzir) [inline Packs/FeedIntel471/Integrations/Intel471Actors/Intel471Actors_test.py:5]:
```suggestion
```

File: Packs/FeedIntel471/Integrations/Intel471Actors/README.md [hunk 17]
```diff
@@ -0,0 +1,68 @@
+"Intel 471's Actors feed is an actor-centric intelligence feature.
+It combines both a field-based intelligence collection and a headquartered-based intelligence analysis component.
+This feed allows getting data out of closed sources (typically referred to as the deep and dark web) where threat actors collaborate, communicate, and plan cyber attacks."
+This integration was integrated and tested with version xx of Intel471 Actors Feed
```
Comment (bakatzir) [inline Packs/FeedIntel471/Integrations/Intel471Actors/README.md:4]:
```suggestion
```

File: Packs/FeedIntel471/Integrations/Intel471Malware/Intel471Malware_test.py [hunk 18]
```diff
@@ -0,0 +1,72 @@
+import pytest
+import Intel471Malware as feed
+# import demistomock as demisto
```
Comment (bakatzir) [inline Packs/FeedIntel471/Integrations/Intel471Malware/Intel471Malware_test.py:3]:
```suggestion
```

File: Packs/FeedIntel471/Integrations/Intel471Malware/README.md [hunk 19]
```diff
@@ -0,0 +1,53 @@
+"Intel471's Malware Intelligence is focused on the provisioning of a high fidelity and timely indicators feed with rich context, TTP information, and malware intelligence reports.
+This feed allows customers to block and gain an understanding of the latest crimeware campaigns and is for those that value timeliness, confidence (little to no false positives), and seek rich context and insight around the attacks they are seeing."
+This integration was integrated and tested with version xx of Intel471 Malware Feed
```
Comment (bakatzir) [inline Packs/FeedIntel471/Integrations/Intel471Malware/README.md:3]:
```suggestion
```

File: Packs/FeedIntel471/pack_metadata.json [hunk 20]
```diff
@@ -0,0 +1,16 @@
+{
+    "name": "Intel471 Feed",
+    "description": "This pack fetches indicators from intel471",
```
Comment (bakatzir) [inline Packs/FeedIntel471/pack_metadata.json:3]: Let's try to give a more detailed desc

File: Packs/FeedJSON/ReleaseNotes/1_1_1.md [hunk 21]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### JSON Feed
+Infrastructure improvements.
```
Comment (bakatzir) [inline Packs/FeedJSON/ReleaseNotes/1_1_1.md:4]:
```suggestion
- Internal code improvements.
```

File: Packs/iDefense/ReleaseNotes/3_0_1.md [hunk 22]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### iDefense Feed
+Infrastructure improvements.
```
Comment (bakatzir) [inline Packs/iDefense/ReleaseNotes/3_0_1.md:4]: - Internal code improvements.

Comment (bakatzir) [inline Packs/iDefense/ReleaseNotes/3_0_1.md:4]: - Internal code improvements.

**Review by bakatzir — CHANGES_REQUESTED** [review]
Please tend to all of the unresolved comments.

**Comment by bakatzir** [comment]
Fails only on the update of the APIMODULE fromversion as expected:
```
Validating backwards compatibility for Packs/ApiModules/Scripts/JSONFeedApiModule/JSONFeedApiModule.yml
Packs/ApiModules/Scripts/JSONFeedApiModule/JSONFeedApiModule.yml: [ST103] - You've added fromversion to an existing file in the system, this is not allowed, please undo.
```
force merging.

**Review by bakatzir — APPROVED** [review]

**Pull request merged**